Formula/aliyun-cli.rb

```diff
@@ -0,0 +1,23 @@
+class AliyunCli < Formula
+  desc "Universal Command-Line Interface for Alibaba Cloud"
+  homepage "https://github.com/aliyun/aliyun-cli"
+  url "https://github.com/aliyun/aliyun-cli/archive/v3.0.13.tar.gz"
+  sha256 "876c4ddc8c4129054196e6420735c600556710df03265cf72291840909a0d9e1"
+
+  depends_on "go"
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["PATH"] = "#{ENV["PATH"]}:#{buildpath}/bin"
+    (buildpath/"src/github.com/aliyun/aliyun-cli").install buildpath.children
+    cd "src/github.com/aliyun/aliyun-cli" do
+      system "make", "metas"
+      system "go", "build", "-o", bin/"aliyun", "-ldflags", "-X 'github.com/aliyun/aliyun-cli/cli.Version=3.0.13'", "main/main.go"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
fixed

Reply:
fixed

Reply:
You can drop the version check all together because it also makes it more difficult to update.

Reply:
ok